### PR TITLE
Date Control Renderer for Angular Material

### PR DESCRIPTION
**Pull request by eneufeld**
~based on #1112~

### CODE REVIEW

File: packages/angular-material/src/controls/date.renderer.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        matInput
+        (dateChange)="onChange($event)"
+        [value]="data"
+        placeholder="{{ description }}"
+        [id]="id"
+        [formControl]="form"
+        [matDatepicker]="myDatepicker"
+    >
+    <mat-datepicker-toggle matSuffix [for]="myDatepicker"></mat-datepicker-toggle>
+    <mat-datepicker #myDatepicker></mat-datepicker>
```
Comment (edgarmueller) [inline packages/angular-material/src/controls/date.renderer.ts]: I'd change the ref name

File: packages/angular-material/test/date-control.spec.ts [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        expect(component.data).toBe('2018-01-01');
+        expect(inputElement.value).toBe('1/1/2018');
+
+        const cells = document.querySelectorAll('.mat-calendar-body-cell');
+        const firstCell: HTMLElement = cells[1] as HTMLElement;
+        firstCell.click();
+        fixture.detectChanges();
+        flush();
+
+        if (inputElement.dispatchEvent) {
```
Comment (edgarmueller) [inline packages/angular-material/test/date-control.spec.ts]: Why do we need the condition here?

File: packages/angular-material/test/date-control.spec.ts [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        const spy = spyOn(component, 'onChange');
+
+        fixture.debugElement.query(By.directive(MatDatepicker)).componentInstance.open();
+        fixture.detectChanges();
+        flush();
+
+        expect(component.data).toBe('2018-01-01');
```
Comment (edgarmueller) [inline packages/angular-material/test/date-control.spec.ts]: I'd remove these expectatations.

File: packages/angular-material/src/controls/date.renderer.ts (hunk 1 above)
Comment (eneufeld) [reply packages/angular-material/src/controls/date.renderer.ts]: do you whether the ref name must be unique across components, or only unique inside a component?

Comment (eneufeld) [reply packages/angular-material/src/controls/date.renderer.ts]: As I understand this article https://blog.angular-university.io/angular-viewchild/#visibilityscopeofviewchildtemplatequeries correctly, then the view ref is bound to the component scope, and thus this should be fine

**Pull request merged**